maintenance/troubleshoot/commands.md

```diff
@@ -0,0 +1,499 @@
+```
+
+```
+NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+calico   True        False         False      27h
+```
+
+#### Get tigera operator details
+
+```bash
+kubectl get tigerastatus
+```
+
+```
+NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+calico   True        False         False      27h
+```
+
+#### Verify component availability
+
+```bash
+kubectl get pod -n tigera-operator
+```
+
+#### View calico nodes
+
+```bash
+kubectl get pod -n calico-system -o wide
+```
+
+#### View cluster information
+
+```bash
+kubectl get installation -o yaml
+```
+
+```yaml
+apiVersion: v1
+items:
+- apiVersion: operator.tigera.io/v1
+  kind: Installation
+  metadata:
+    - apiVersion: operator.tigera.io/v1
+ spec:
+    calicoNetwork:
+      bgp: Enabled
+      hostPorts: Enabled
+      ipPools:
+      - blockSize: 26
+        cidr: 10.48.0.0/16
+        encapsulation: VXLANCrossSubnet
+        natOutgoing: Enabled
+        nodeSelector: all()
+      multiInterfaceMode: None
+      nodeAddressAutodetectionV4:
+        firstFound: true
+    cni:
+      ipam:
+        type: Calico
+      type: Calico
+```
+
+#### Run commands across multiple nodes
+
+Export THE_COMMAND_TO_RUN=date && for calinode in:
+
+```bash
+kubectl get pod -o wide -n calico-system | grep calico-node | awk '{print $1}'`; do echo $calinode; echo "-----"; 
+```
+
+```bash
+kubectl exec -n calico-system $calinode -- $THE_COMMAND_TO_RUN; printf "\n"; done calico-node-8xfmx
+```
+
+```
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-9t8s7
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-9cjhw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-cb7ff
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-qoxvw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-wm5m2
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+```
+
+#### View pod info of pod that is not running
+
+```bash
+kubectl describe pods `<pod_name>`  -n `<namespace> `
+```
+
+```bash
+kubectl describe pods busybox -n default
+```
+
+```
+Events:
+  Type    Reason     Age   From                   Message
+  ----    ------     ----  ----                   -------
+  Normal  Scheduled  21s   default-scheduler      Successfully assigned default/busybox to ip-10-0-0-11
+  Normal  Pulling    20s   kubelet, ip-10-0-0-11  Pulling image "busybox"
+  Normal  Pulled     19s   kubelet, ip-10-0-0-11  Successfully pulled image "busybox"
+  Normal  Created    19s   kubelet, ip-10-0-0-11  Created container busybox
+  Normal  Started    18s   kubelet, ip-10-0-0-11  Started container busybox
+```
+
+#### View logs of a pod that is not running
```

Comment:
This won't work on all pods - it will only work on pods that are running (or have run and subsequently failed / exited).

Probably should just be "View logs of a pod"

Reply:
Fixed.